go/proxyd/server.go

```diff
@@ -108,23 +127,45 @@ func (s *Server) HandleWS(w http.ResponseWriter, r *http.Request) {
 	proxier, err := s.backends.ProxyWS(clientConn)
 	if err != nil {
 		if errors.Is(err, ErrNoBackends) {
-			RecordUnserviceableRequest(RPCRequestSourceWS)
+			RecordUnserviceableRequest(ctx, RPCRequestSourceWS)
 		}
 		log.Error("error dialing ws backend", "err", err)
 		clientConn.Close()
 		return
 	}
 
-	activeClientWsConnsGauge.Inc()
+	activeClientWsConnsGauge.WithLabelValues(GetAuthCtx(ctx)).Inc()
 	go func() {
 		// Below call blocks so run it in a goroutine.
-		if err := proxier.Proxy(); err != nil {
+		if err := proxier.Proxy(ctx); err != nil {
 			log.Error("error proxying websocket", "err", err)
 		}
-		activeClientWsConnsGauge.Dec()
+		activeClientWsConnsGauge.WithLabelValues(GetAuthCtx(ctx)).Dec()
 	}()
 }
 
+func (s *Server) authenticate(w http.ResponseWriter, r *http.Request) context.Context {
+	vars := mux.Vars(r)
+	authorization := vars["authorization"]
+
+	if s.authenticatedPaths == nil {
+		// handle the edge case where auth is disabled
+		// but someone sends in an auth key anyway
+		if authorization != "" {
```

Comment:
Should we ignore the auth key in this case?

Reply:
I'm hesitant to ignore this case because it creates an implied contract when auth is disabled, i.e. that we'll accept any auth key in the URL rather than requiring none.